src/atoms/affine/stack.jl

```diff
@@ -111,27 +111,32 @@ function conic_form!(x::HcatAtom, unique_conic_forms::UniqueConicForms=UniqueCon
 end
 
 hcat(args::AbstractExpr...) = HcatAtom(args...)
-hcat(args::AbstractExprOrValue...) = HcatAtom([convert(AbstractExpr, arg) for arg in args]...)
-hcat(args::Value...) = Base.cat(2, args...)
+hcat(args::AbstractExprOrValue...) = HcatAtom(map(arg -> convert(AbstractExpr, arg), args)...)
+hcat(args::Value...) = Base.cat(args..., dims=2)
 
 
 # TODO: implement vertical concatenation in a more efficient way
-vcat(args::AbstractExpr...) = transpose(HcatAtom([transpose(arg) for arg in args]...))
-vcat(args::AbstractExprOrValue...) = transpose(HcatAtom([transpose(convert(AbstractExpr, arg)) for arg in args]...))
-vcat(args::Value...) = Base.cat(1, args...) # Note: this makes general vcat slower for anyone using Convex...
-
-
-Base.vect{T<:AbstractExpr}(args::T...) = transpose(HcatAtom([transpose(arg) for arg in args]...))
-Base.vect(args::AbstractExpr...) = transpose(HcatAtom([transpose(arg) for arg in args]...))
-Base.vect(args::AbstractExprOrValue...) = transpose(HcatAtom([transpose(convert(AbstractExpr,arg)) for arg in args]...))
-if Base._oldstyle_array_vcat_
-  Base.vect(args::Value...) = Base.vcat(args...)
-  # This is ugly, because the method redefines simple cases like [1,2,3]
-        
-else
-    function Base.vect(args::Value...)
-        T = Base.promote_typeof(args...)
-        return copy!(Array{T}(length(args)), args)
-
+vcat(args::AbstractExpr...) = transpose(HcatAtom(map(transpose, args)...))
+vcat(args::AbstractExprOrValue...) = transpose(HcatAtom(map(arg -> transpose(convert(AbstractExpr, arg)), args)...))
+vcat(args::Value...) = Base.cat(args..., dims=1) # Note: this makes general vcat slower for anyone using Convex...
+
+
+function hvcat(rows::Tuple{Vararg{Int}}, args::AbstractExprOrValue...)
+    nbr = length(rows)
+    rs = Vector{Any}(undef, nbr)
+    a = 1
+    for i = 1:nbr
+        rs[i] = hcat(args[a:a-1+rows[i]]...)
+        a += rows[i]
     end
+    return vcat(rs...)
+end
+
+
+Base.vect(args::T...) where {T<:AbstractExpr} = transpose(HcatAtom(map(transpose, args)...))
+Base.vect(args::AbstractExpr...) = transpose(HcatAtom(map(transpose, args)...))
+Base.vect(args::AbstractExprOrValue...) = transpose(HcatAtom(map(arg -> transpose(convert(AbstractExpr, arg)), args)...))
+
+function Base.vect(args::Value...)
```

Comment:
This similarly seems unnecessary, as the Base methods work.

Reply:
Same thing like above.